server/container_create.go

```diff
@@ -571,9 +571,17 @@ func (s *Server) setupSeccomp(ctx context.Context, specgen *generate.Generator,
 			specgen.Config.Linux.Seccomp = nil
 			return nil
 		}
+		// default to SeccompProfileRuntimeDefault if user sets UseDefaultWhenEmpty
 		profile = k8sV1.SeccompProfileRuntimeDefault
 	}
+	// kubelet forces sandbox to run as `runtime/default` , if !UseDefaultWhenEmpty we consider it as unconfined if Seccomp disabled
+	// https://github.com/kubernetes/kubernetes/blob/12d9183da03d86c65f9f17e3e28be3c7c18ed22a/pkg/kubelet/kuberuntime/kuberuntime_sandbox.go#L162-L163
 	if s.Config().Seccomp().IsDisabled() {
+		if profile == k8sV1.SeccompProfileRuntimeDefault && !s.Config().Seccomp().UseDefaultWhenEmpty() {
```

Comment:
I think we can apply this independently from the `UseDefaultWhenEmpty` option.
```suggestion
		if profile == k8sV1.SeccompProfileRuntimeDefault {
```

Reply:
done